crates/python/src/py_sync.rs

```diff
@@ -50,6 +53,12 @@ macro_rules! py_async {
 /// named after it that can be used to invoke either
 /// blocking or async variants of the same function.
 ///
+/// The given function will be spawned on a Rust event loop
+/// this means functions like [`pyo3::Python::with_gil`]
+/// should not be used, as acquiring Python's global
+/// interpreter lock from a Rust runtime
+/// isn't possible.
+///
```

Comment:
I don't love that this is only "enforced" by a comment, interested in other ways of handling it.

Reply:
Ideally it'd be passed through to the inner function - I wonder if https://docs.rs/pyo3/latest/pyo3/marker/struct.Python.html#method.assume_gil_acquired can still be used instead, if only in dire circumstances. Since it's not something that can be enforced statically, I think a warning comment is about as much as you can do.

Reply:
Yeah, I confirmed that the inner function can't take it as an argument because it's async. I'll keep it in mind, but since this side effect isn't new to this PR, I'll go ahead and merge for now.